verilog/tools/kythe/kythe_facts.cc

```diff
@@ -53,6 +53,14 @@ std::string Signature::ToBase64() const {
   return absl::Base64Escape(ToString());
 }
 
+bool VName::operator==(const VName& other) const {
+  return this->ToString() == other.ToString();
+}
+
+bool VName::operator<(const VName& other) const {
+  return this->ToString() < other.ToString();
```

Comment:
(No need to fix for now)

While this comparison works, it's far from optimal. ToString() creates a new string just for a sake of comparison. While comparing two short strings is fast, allocating them isn't.

Doing a comparison by hand might make sense here.
Also, keep in mind that manually writing comparators is error prone (seems simple, but it's easy to introduce a mistake). Using tricks like with std::tie[1] might be useful.

[1] http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n3950.html

Reply:
Agreed on tuple-comparison recommendation.  (At least leave a TODO.)